compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -160,20 +160,34 @@ object SymDenotations {
      */
     private[dotc] final def flagsUNSAFE: FlagSet = myFlags
 
+    private def setMyFlags(flags: FlagSet) =
+      lazy val immutableFlags = if myInfo.isInstanceOf[SymbolLoader] then AfterLoadFlags else FromStartFlags
```

Comment:
Every local lazy val is a boxed object, should be avoided in performance sensitive code like this.